### ISSUE TITLE
Requests 2.32.0 Not supported URL scheme http+unix

### ISSUE THREAD

**Issue by cartertinney** [msabramo/requests-unixsocket#73]
Latest security fix for requests has broken requests-unixsocket. [It appears they do not plan to fix the breakage](https://github.com/psf/requests/issues/6707), and are suggesting it be the responsibility of consuming libraries.

Will there be a fix coming for this? Or will PRs for fixing it be accepted/released?

**Comment by brasse** [comment]
Would the same fix work here?

https://github.com/docker/docker-py/pull/3257

**Comment by cartertinney** [comment]
@brasse Yes, that is my understanding that that would be all that is required.

**Comment by crgwbr** [comment]
FYI to all: since this project seems to be abandoned, but its longevity is important to my team, we've forked the project as `requests-unixsocket2`. It should be a drop in replacement for this package.

- PyPI: https://pypi.org/project/requests-unixsocket2/0.4.0/
- Repository: https://gitlab.com/thelabnyc/requests-unixsocket2

We've migrated the fix for this issue there, merged it, and released to PyPI as part of [v0.4.0](https://pypi.org/project/requests-unixsocket2/0.4.0/).

**Comment by webknjaz** [comment]
xref #72

**Comment by ai-nikolai** [comment]
Encountering this issue with:
```
requests==2.32.3
requests-unixsocket==0.3.0
```

**Comment by mcepl** [comment]
If anybody is interested in (slightly more complicated) patch against urllib3 2.31.0.

[requests230.patch](https://github.com/user-attachments/files/21552680/requests230.patch.txt)

**Comment by webknjaz** [comment]
@mcepl if you want to send a PR, I can review it.

**Comment by webknjaz** [comment]
@mcepl you can only create PRs against branches, nor tags.
